robottelo/decorators.py

```diff
@@ -106,29 +94,48 @@ def test_hostgroup_create():
 
     """
     allowed_project_modes = ('sat', 'sam')
-
-    # Validate project value
     project = project.lower()
-    settings_project = settings.project.lower() if settings.project else 'sat'
-    if project not in allowed_project_modes:
-        raise ProjectModeError(
-            '"{0}" is not a project mode. The allowed project modes are: {1}'
-            .format(project, allowed_project_modes)
-        )
 
-    # If robottelo.properties not present or does not specify a project use sat
-    if settings_project not in allowed_project_modes:
-        raise ProjectModeError(
-            '"{0}" is not an acceptable "[robottelo] project" value in '
-            'robottelo.properties file. The allowed project modes are: {1}'
-            .format(settings.project, allowed_project_modes)
-        )
+    def decorator(func):
```

Comment:
Without this the run_only_on cannot receive a parameter. Previous was being returned the skipIf which is a function that receive a function as argument. And because this the "internal" implementation of a decorator that receives arguments was implicit.